File: src/FloodFill.java
public class FloodFill {
    private int[][] image;
    private int height;
    private int width;

    public int[][] floodFill(int[][] image, int sr, int sc, int newColor) {
        this.image = image;
        height = image.length;
        width = image[0].length;
        if (image[sr][sc] != newColor) {
            colorPixel(sr, sc, image[sr][sc], newColor);
        }
        return image;
    }

    private void colorPixel(int row, int col, int oldColor, int newColor) {
        if (image[row][col] != newColor) {
            image[row][col] = newColor;
            if (row - 1 >= 0 && image[row - 1][col] == oldColor) colorPixel(row - 1, col, oldColor, newColor);
            if (col + 1 < width && image[row][col + 1] == oldColor) colorPixel(row, col + 1, oldColor, newColor);
            if (row + 1 < height && image[row + 1][col] == oldColor) colorPixel(row + 1, col, oldColor, newColor);
            if (col - 1 >= 0 && image[row][col - 1] == oldColor) colorPixel(row, col - 1, oldColor, newColor);
        }
    }
}

File: python/flood_fill.py
from typing import List


class Solution:

    def floodFill(self, image: List[List[int]], sr: int, sc: int, newColor: int) -> List[List[int]]:
        self.image = image
        self.height = len(image)
        self.width = len(image[0])
        self.newColor = newColor
        if image[sr][sc] != newColor:
            self.oldColor = image[sr][sc]
            self.colorPixel(row=sr, column=sc)
        return image

    def colorPixel(self, row: int, column: int) -> None:
        if self.image[row][column] != self.newColor:
            self.image[row][column] = self.newColor
            if row - 1 >= 0 and self.image[row - 1][column] == self.oldColor: self.colorPixel(row - 1, column)
            if column + 1 < self.width and self.image[row][column + 1] == self.oldColor: self.colorPixel(row, column + 1)
            if row + 1 < self.height and self.image[row + 1][column] == self.oldColor: self.colorPixel(row + 1, column)
            if column - 1 >= 0 and self.image[row][column - 1] == self.oldColor: self.colorPixel(row, column - 1)
